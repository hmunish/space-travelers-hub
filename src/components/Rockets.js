import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { fetchRockets } from '../redux/rockets/rocketsSlice';

const Rockets = () => {
  const rocketsData = useSelector((state) => state.rocketsList.value);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(fetchRockets());
  }, [dispatch]);

  return (
    <ul>
      {rocketsData.map((rocket) => (
        <li key={rocket.id}>{rocket.cost_per_launch}</li>
      ))}
    </ul>
  );
};

export default Rockets;
