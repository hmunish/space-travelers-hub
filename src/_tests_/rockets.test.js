import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import Rockets from '../components/Rockets';

const mockStore = configureStore([]);

describe('test rockets component', () => {
  const mockRockets = [
    {
      id: 1,
      name: 'Falcon',
      flickrImages: '',
      description: 'Falcon description',
      reserved: false,
    },
    {
      id: 2,
      name: 'Starship',
      flickrImages: '',
      description: 'Starship description',
      reserved: true,
    },
  ];

  const store = mockStore({
    rocketsList: {
      value: mockRockets,
      loading: false,
      errors: null,
    },
  });
  it('should match the snapshot', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Rockets />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
