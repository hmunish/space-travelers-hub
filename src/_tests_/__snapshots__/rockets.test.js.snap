// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test rockets component should match the snapshot 1`] = `
<ul
  className="rocketsList"
>
  <li
    className="rocketCard"
    data-id={1}
  >
    <img
      alt="Falcon"
      className="rocketCard"
      src=""
    />
    <div
      className="wrapper"
    >
      <h3
        className="rocketCard"
      >
        Falcon
      </h3>
      <p
        className="rocketCard"
      >
         
        Falcon description
      </p>
      <button
        className="rocketCard"
        data-index-no={0}
        onClick={[Function]}
        type="button"
      >
        Reserve Rocket
      </button>
    </div>
  </li>
  <li
    className="rocketCard"
    data-id={2}
  >
    <img
      alt="Starship"
      className="rocketCard"
      src=""
    />
    <div
      className="wrapper"
    >
      <h3
        className="rocketCard"
      >
        Starship
      </h3>
      <p
        className="rocketCard"
      >
        <span
          className="reservedTag"
        >
          Reserved
        </span>
         
        Starship description
      </p>
      <button
        className="rocketCard cancelBtn"
        data-index-no={1}
        onClick={[Function]}
        type="button"
      >
        Cancel Reservation
      </button>
    </div>
  </li>
</ul>
`;
